src/channel_worker.erl

```diff
@@ -65,7 +66,7 @@ start(Peer_endpoint, Swarm_options) ->
 %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
 %% @doc Stops the server, given a channel.
 
--spec stop(ppspp_channel:channel()) -> ok | {error, any()}.
+-spec stop({channel, ppspp_channel:channel()}) -> ok | {error, any()}.
```

Comment:
this doesn't feel right here -- either we're passing in an opaque object or we're not. Is there a reason not to use `#{channel => Channel}` as an opaque type from `ppspp_channel()` ?